### PR TITLE
removed unwanted data fields from API responses

### PR DESCRIPTION
**Pull request by thevinitgupta**
## Describe your changes
Removed the unwanted data being send to the frontend from the API
Also, replaced with success message where needed.

## Issue ticket number and link - If Any
#217 

## Checklist before requesting a review

- [x] I have performed a self-review of my code.

- [x] Followed the repository's [Contributing Guidelines](https://github.com/dvstechlabs/Noteslify/blob/main/CONTRIBUTING.md).

- [x] I ran the app and tested it locally to verify that it works as expected.

- [x] I have starred the repository.

### CODE REVIEW

**Review by devarshishimpi — APPROVED** [review]
LGTM! Make sure to commit the package-lock.json file as well. I know it's not recommended to add, but seems like we need for particular versioning

**Comment by devarshishimpi** [comment]
@Gauravdarkslayer @abhinandanwadwa Kindly merge the PR after any of the second reviews completes!

**Comment by Gauravdarkslayer** [comment]
@thevinitgupta @devarshishimpi I guess the same thing goes for the folder section as well.

File: apps/web/backend/routes/notes.js [hunk 1]
```diff
@@ -107,12 +107,19 @@ router.delete('/deletenotepermanently/:id', fetchuser, async (req, res) => {
 router.get('/getallnotes', fetchuser, async (req, res) => {
     try {
         const allNotes = await NoteSchema.find({ authorId: req.user.id, isDeleted: false }).sort({ createdAt: -1 });
+        let notesList = [];
         for (let index = 0; index < allNotes.length; index++) {
             const element = allNotes[index];
             element.title =  helper.decrypt(element.title, element.secretKey);
             element.description =  helper.decrypt(element.description, element.secretKey);
+            notesList.push({
```
Comment (Gauravdarkslayer) [inline apps/web/backend/routes/notes.js:115]: Without pushing it to a new array, the unwanted key can also be deleted.

**Review by Gauravdarkslayer — COMMENTED** [review]
Without pushing it to a new array, the unwanted key can also be deleted, so that we don't need an extra array.

File: apps/web/backend/routes/notes.js (hunk 1 above)
Comment (Gauravdarkslayer) [reply apps/web/backend/routes/notes.js:115]: Need to implement the same for the folder section routes.

**Comment by thevinitgupta** [comment]
It is a `const` array. That is why I added a new array to return the required data.

**Review by abhinandanwadwa — CHANGES_REQUESTED** [review]
https://github.com/dvstechlabs/Noteslify/blob/1bcd840142bb9fd27146771dec540719a5ead527/apps/web/backend/routes/folders.js#L37

https://github.com/dvstechlabs/Noteslify/blob/1bcd840142bb9fd27146771dec540719a5ead527/apps/web/backend/routes/folders.js#L186

https://github.com/dvstechlabs/Noteslify/blob/1bcd840142bb9fd27146771dec540719a5ead527/apps/web/backend/routes/folders.js#L211

Here are few more

**Comment by thevinitgupta** [comment]
@abhinandanwadwa shall I work on the folders route or wait for @devarshishimpi ?

**Comment by abhinandanwadwa** [comment]
@thevinitgupta Go ahead with folder routes

**Comment by thevinitgupta** [comment]
@abhinandanwadwa updated the folders routes.

**Review by abhinandanwadwa — APPROVED** [review]

**Review by Gauravdarkslayer — COMMENTED** [review]
Please use the delete key method I've mentioned earlier rather than creating a new array of objects.

**Review by devarshishimpi — APPROVED** [review]

**Comment by devarshishimpi** [comment]
> Please use the delete key method I've mentioned earlier rather than creating a new array of objects.

@Gauravdarkslayer Is the required changes necessary or could we merge the PR?

**Comment by Gauravdarkslayer** [comment]
They are not necessary but its required for performance & space related optimizations.
@devarshishimpi

**Comment by devarshishimpi** [comment]
> They are not necessary but its required for performance & space related optimizations. @devarshishimpi

Great, @thevinitgupta Could you complete the required change as well?

**Comment by thevinitgupta** [comment]
@Gauravdarkslayer can I use the `select` method of the Schema prototype in this way : 
```javascript
const allNotes = await NoteSchema.find({ authorId: req.user.id, isDeleted: false }).select({_id : 1, title : 1, description : 1, createdAt : 1, secretKey : 1}).sort({ createdAt: -1 });
```

This way, we'll only need to delete the `secretKey`.

**Comment by Gauravdarkslayer** [comment]
@thevinitgupta Yes you can do this, but please make sure to be consistent throughout the application. Like if you are doing this for notes routes and do the same for folder routes as well.

**Comment by thevinitgupta** [comment]
Sure, I will do that.

**Comment by thevinitgupta** [comment]
I have added the `select` query and used the `delete` key method for the notes and folders routes.

**Review by Gauravdarkslayer — APPROVED** [review]
LGTM !

**Pull request merged**